precompile/modules/initia_stdlib/sources/vip/weight_vote.move

```diff
@@ -1177,6 +1177,48 @@ module initia_std::vip_weight_vote {
         }
     }
 
+    #[view]
+    public fun get_challenge_by_stage(stage: u64): vector<ChallengeResponse> acquires ModuleStore {
+        let module_store = borrow_global<ModuleStore>(@initia_std);
+        let iter = table::iter(
+            &module_store.challenges,
+            option::none(),
+            option::none(),
+            1
+        );
+
+        let challenge_responses = vector::empty<ChallengeResponse>();
+        loop {
```

Comment:
how about storing challenge key to stage data as `Option` to avoid iteration?

Reply:
seems challenge query cost will not be expensive i guess